### ISSUE TITLE
`@typescript-eslint/no-redundant-type-constituents` false positive with `$$Generic`

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#541]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

v8.44.0

### What version of `eslint-plugin-svelte` are you using?

2.32.2

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
  "rules": {
      "@typescript-eslint/no-redundant-type-constituents": "error"
  }
}
```
</details>

```svelte
<script>
  type KeyType = $$Generic<number | string>

  export let selected: KeyType | null
</script>
```

### What did you expect to happen?

No error

### What actually happened?

```
4:24 error  'any' overrides all other types in this union type  @typescript-eslint/no-redundant-type-constituents
```

### Link to **GitHub Repo** with Minimal Reproducible Example

-

**Comment by marekdedic** [comment]
Generics are handled differently in Svelte 5, so closing this as irrelevant.
